tlslite/messages.py

```diff
@@ -1765,3 +1759,82 @@ def parse(self, p):
 
     def write(self):
         return self.bytes
+
+
+class Heartbeat(object):
+    """
+    Handling Heartbeat messages from RFC 6520
+
+    @type message_type: int
+    @ivar message_type: type of message (response or request)
+
+    @type payload: bytearray
+    @ivar payload: payload
+
+    @type padding: bytearray
+    @ivar padding: random padding of selected length
+    """
+
+    def __init__(self):
+        self.contentType = ContentType.heartbeat
+        self.message_type = 0
+        self.payload = bytearray(0)
+        self.padding = bytearray(0)
+
+    def create(self, message_type, payload, padding_length):
+        """Create heartbeat request or response with selected parameters"""
+        self.message_type = message_type
+        self.payload = payload
+        self.padding = getRandomBytes(padding_length)
+        return self
+
+    def createResponse(self):
+        """Creates heartbeat response based on request."""
+        heartbeat_response = Heartbeat().create(
+            HeartbeatMessageType.heartbeat_response,
+            self.payload,
+            16)
+        return heartbeat_response
+
+    def checkResponse(self, callback=None):
```

Comment:
code in `messages.py` and `extensions.py` is just parsing and decoding values, it does not perform any kind of validation or verification outside of checking if the message is well formed.

For now I'm putting that code in [tlsconnection.py](https://github.com/tomato42/tlslite-ng/blob/master/tlslite/tlsconnection.py#L1486), later on it will need to be moved in a separate module, but definitely not here.